File: apps/desktop/src/main/attention-policy.ts
import type { AppConfig } from "../shared/config";

export interface AttentionRequest {
  readonly durationMs: number;
  readonly kind: "none" | "edge" | "overlay";
  readonly placement?: "center" | "corner";
}

export const attentionRequestForConfig = (config: AppConfig, durationMs = config.attentionDurationMs): AttentionRequest => {
  switch (config.attentionMode) {
    case "off":
      return { kind: "none", durationMs };
    case "panel-pulse":
      return { kind: "edge", durationMs };
    case "corner-overlay":
      return { kind: "overlay", durationMs, placement: "corner" };
    case "center-overlay":
      return { kind: "overlay", durationMs, placement: "center" };
  }
};

File: apps/macos-native/Sources/Core/AttentionPolicy.swift
import Foundation

/// 提醒方式映射（对照 src/main/attention-policy.ts:3-20）。
public enum AttentionRequestKind: Equatable, Sendable {
    case none
    case edge
    case overlay(placement: AttentionPlacement)
}

public enum AttentionPlacement: String, Equatable, Sendable {
    case corner
    case center
    case edge
}

public enum AttentionPolicy {
    public static func request(for config: AppConfig, durationMs: Int? = nil) -> (kind: AttentionRequestKind, durationMs: Int) {
        let duration = durationMs ?? config.attentionDurationMs
        switch config.attentionMode {
        case .off:
            return (.none, duration)
        case .panelPulse:
            return (.edge, duration)
        case .cornerOverlay:
            return (.overlay(placement: .corner), duration)
        case .centerOverlay:
            return (.overlay(placement: .center), duration)
        }
    }
}
